great_expectations/validator/validator.py

```diff
@@ -1511,6 +1492,37 @@ def validate(  # noqa: C901, PLR0912, PLR0913, PLR0915
 
         return result
 
+    def process_expectations_for_validation(
+        self,
+        expectation_configurations: list[ExpectationConfiguration],
+        evaluation_parameters: Optional[dict[str, Any]] = None,
+    ) -> list[ExpectationConfiguration]:
+        """Substitute evaluation parameters into the provided expectations and sort by column."""
+        columns: dict[Any, list[ExpectationConfiguration]] = {}
+
+        for expectation in expectation_configurations:
+            expectation.process_evaluation_parameters(
+                evaluation_parameters=evaluation_parameters,
+                interactive_evaluation=self.interactive_evaluation,
+                data_context=self._data_context,
+            )
+            if "column" in expectation.kwargs and isinstance(
+                expectation.kwargs["column"], Hashable
+            ):
+                column = expectation.kwargs["column"]
+            else:
+                # noinspection SpellCheckingInspection
+                column = "_nocolumn"
```

Comment:
Not any fault of this PR but I don't love this magic string. Can we see where else we use it and perhaps make it a constant/enum?

Reply:
Yeah, I was confused when I first saw that. I originally kept it so that this was just a strict cut/paste, but the lift is low to update this. Will do.